### DISCUSSION TITLE
[Q&A] Problem in using default value for the parameter

### DISCUSSION THREAD

**Discussion by Sumit-Negi-github** [fluent/fluentd#4713] [Q&A]
### What is a problem?

**Objective:** To use default value of param if param is not provided in config. 
I am writing it like this - 
```
config_param :dummy_param, :string, default: 'dummy_value'
def configure(conf)
    super
    @log.info("conf['dummy_param']=#{conf['dummy_param']}")
```

If i dont provide the param in the config, this is not printing dummy_value in the logs. It seems it is not getting picked up.

This code is in custom output plugin where i have inherited `Fluent::Plugin::Output`

Any help on where i am going wrong?

TIA.

### Describe the configuration of Fluentd

Not needed.

### Environment

```markdown
- Fluentd version: fluentd 1.16.5
- TD Agent version:
- Fluent Package version:
- Docker image (tag):
- Operating system: OL8
- Kernel version:
```

**Comment by daipom** [comment]
@Sumit-Negi-github 
We can use the parameters by `@...` after `super` in `configure`.

```rb
log.info("conf['dummy_param']=#{@dummy_param}")
```

example: https://docs.fluentd.org/plugin-development/api-config-types#data-types

**Reply by Sumit-Negi-github** [reply]
By using @dummy_param, that is printing the correct value in the logs. But is this actually getting considered for default value by fluentd internally. 
I want to make use of the @id parameter provided by fluentd. If i dont have @id parameter in config, i want to make sure the default value for it gets considered. So logs should be printed in the log file with corresponding @id value of that config section. But they are not printing with @id.
